beater/server_test.go

```diff
@@ -45,7 +45,8 @@ func TestMain(m *testing.M) {
 }
 
 func TestServerOk(t *testing.T) {
-	apm, teardown := setupServer(t, nil)
+	apm, teardown, err := setupServer(t, nil)
+	assert.NoError(t, err)
```

Comment:
We recently introduced using [testify/require](https://godoc.org/github.com/stretchr/testify/require) for checks that should fail fast and prevent the test from being run any further. I think this might also make sense for some of the `NoError` checks here. 

(see conversation around introducing _require_ https://github.com/elastic/apm-server/pull/1015#discussion_r197250826)

Reply:
done here 3be0983417f7dc4b5427cab78ec8f4f5da4a61f7